### PR TITLE
bugfix remove duplicated data.

### PR DESCRIPTION
**Pull request by Jackie9527**
There are some problems when using assimp to process some fbx files exported from UE.
After analysis, it is confirmed that the problems are caused by duplicate data, which could be solved by eliminating duplicate data.

### CODE REVIEW

File: code/AssetLib/FBX/FBXConverter.cpp [hunk 1]
```diff
@@ -1176,15 +1176,23 @@ unsigned int FBXConverter::ConvertMeshSingleMaterial(const MeshGeometry &mesh, c
...
+                const auto &curIndices = shapeGeometry->GetIndices();
                 //losing channel name if using shapeGeometry->Name()
-                animMesh->mName.Set(FixAnimMeshName(blendShapeChannel->Name()));
+                // if blendShapeChannel Name is empty or don't have a ".", add geoMetryName;
+                auto aniName = FixAnimMeshName(blendShapeChannel->Name());
+                auto geoMetryName = FixAnimMeshName(shapeGeometry->Name());
+                if (aniName.empty()) {
+                    aniName = geoMetryName;
+                }
+                else if (aniName.find(".") == aniName.npos) {
```
Comment (Skylion007) [inline code/AssetLib/FBX/FBXConverter.cpp:1192]:
```suggestion
                else if (aniName.find('.') == aniName.npos) {
```

File: code/AssetLib/FBX/FBXConverter.cpp [hunk 2]
```diff
@@ -1176,15 +1176,23 @@ unsigned int FBXConverter::ConvertMeshSingleMaterial(const MeshGeometry &mesh, c
...
+                    aniName = aniName + "." + geoMetryName;
```
Comment (Skylion007) [inline code/AssetLib/FBX/FBXConverter.cpp:1193]:
```suggestion
                    aniName += "." + geoMetryName;
```
is also a more efficient overload

File: code/AssetLib/FBX/FBXConverter.cpp (hunk 1 above)
Comment (Jackie9527) [reply code/AssetLib/FBX/FBXConverter.cpp]: done.

File: code/AssetLib/FBX/FBXConverter.cpp (hunk 2 above)
Comment (Jackie9527) [reply code/AssetLib/FBX/FBXConverter.cpp]: done.

**Review by kimkulling — APPROVED** [review]
Looks fine.

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Comment by turol** [comment]
https://github.com/assimp/assimp/pull/6146 wants to partially revert this. @Jackie9527 Do you have a testcase file for this?

**Pull request merged**